File: spark-lab/src/main/scala/org/cg/spark_lab/TestSpark.scala
package org.cg.spark_lab


import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object TestSpark {

	def main(args: Array[String]) {

		val conf = new SparkConf().setAppName("scala_spark_test").setMaster("local[2]")
		val sc = new SparkContext(conf)
		val data = Array(1, 2, 3, 4, 5)
        val distData = sc.parallelize(data)
		distData.collect().foreach(println)

	}


}

File: spark-lab-java8/src/main/java/spark_lab_java8/TestSpark.java
package spark_lab_java8;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class TestSpark {
	
	public static void main(String args[]){
		
		SparkConf conf = new SparkConf().setAppName("sparkTest").setMaster("local[2]");
		JavaSparkContext sc = new JavaSparkContext(conf);
		
		JavaRDD<String> lines = sc.textFile("data/data.txt");
		JavaRDD<Integer> lineLengths = lines.map(s -> s.length());
		int totalLength = lineLengths.reduce((a, b) -> a + b);
		System.out.println(totalLength);
		
	}

}
